imap_processing/ultra/l1c/l1c_lookup_utils.py

```diff
@@ -243,6 +250,9 @@ def get_spacecraft_pointing_lookup_tables(
     for_indices_by_spin_phase = np.nan_to_num(index_grid[:, 2:], nan=0).astype(
         bool
     )  # Shape (npix, 15000)
+    boundary_scale_factors = xr.DataArray(
+        boundary_scale_factors, dims=("pixel", "spin_phase_step")
+    )
```

Comment:
There is a `to_xarray()` method on pandas dataframes too. But, that might actually be even more confusing, so just putting this out there in case you'd find that easier. This seems good too though.
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_xarray.html